wallet-api/src/Wallet/API.hs

```diff
@@ -298,6 +304,10 @@ throwInsufficientFundsError = throwError . InsufficientFunds
 throwOtherError :: MonadError WalletAPIError m => Text -> m a
 throwOtherError = throwError . OtherError
 
+register :: (WalletAPI m, Monad m) => EventTrigger -> EventHandler m -> m ()
+register t h = registerOnce t h' where
```

Comment:
Interesting: this parallels something Mario and I discussed with callbacks. There are a few ways of doing this:
- Callback is registered until it is explicitly unregistered.
- Callback is registered but has access to its own handle so can unregister itself.
- Callback is registered to run only once but can reschedule iteslf (this version).

A downside of this is that I think it's awkward to allow the user to cancel a trigger. If `registerOnce` gives you a fresh id for the trigger, then we have no way to pass it back to the user when we re-register.

Ideally I think we want ooth of:
- Triggers can run for as long as they want (i.e. they can cancel/reschedule themselves)
- Triggers can be cancelled by the user throughout their lifetimes.

Reply:
You could add a flag `cancelled` to the contract state, and write a trigger that runs every slot and only reschedules itself if it hasn't been cancelled. But that would not be as efficient as handles.